### PR TITLE
[SourceKit] Add an option to sort completion result

### PR DESCRIPTION
**Pull request by rintaro**
Defaults to false

rdar://problem/60799439

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
https://github.com/apple/sourcekit-lsp/pull/258
@swift-ci Please test

File: test/SourceKit/CodeComplete/complete_sort_order.swift [hunk 1]
```diff
@@ -9,23 +9,30 @@ func test() {
 
 // XFAIL: broken_std_regex
 // RUN: %sourcekitd-test -req=complete -req-opts=hidelowpriority=0 -pos=7:1 %s -- %s > %t.orig
-// RUN: %FileCheck -check-prefix=NAME %s < %t.orig
-// Make sure the order is as below, foo(Int) should come before foo(String).
+// RUN: %sourcekitd-test -req=complete -req-opts=hidelowpriority=0,sort.byname=0 -pos=7:1 %s -- %s > %t.orig.off
+// RUN: %FileCheck -check-prefix=NAME_SORTED %s < %t.orig
+// RUN: %FileCheck -check-prefix=NAME_UNSORTED %s < %t.orig.off
```
Comment (akyrtzi) [inline test/SourceKit/CodeComplete/complete_sort_order.swift]: How about adding a `diff` check for `%t.orig` vs `%t.orig.off` to ensure we got different behavior?
I suspect the test will still pass even if both invocations end up sorting the results.

Comment (rintaro) [reply test/SourceKit/CodeComplete/complete_sort_order.swift]: > I suspect the test will still pass even if both invocations end up sorting the results.

Right.

I will add the `diff` test.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by akyrtzi — APPROVED** [review]

**Comment by rintaro** [comment]
https://github.com/apple/sourcekit-lsp/pull/258
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/apple/sourcekit-lsp/pull/258
@swift-ci Please smoke test Linux

**Pull request merged**